### ISSUE TITLE
Do not expose added members via Object.keys

### ISSUE THREAD

**Issue by TheSavior** [speedskater/babel-plugin-rewire#22]
babel-plugin-rewire should make sure that if you Object.keys() on a module that you don't see the added functions in the array:

```
[ 'myKey1',
  'myKey2',
  '__GetDependency__',
  '__get__',
  '__Rewire__',
  '__set__',
  '__ResetDependency__' ]
```

I fixed this in my package [`rewire-global`](https://github.com/TheSavior/rewire-global/blob/master/index.js)

And we just fixed this in rewireify: https://github.com/i-like-robots/rewireify/pull/14

**Comment by TheSavior** [comment]
Note that this is also part of @jhnns 's Rewire proper. You can see the Object.define there too: https://github.com/jhnns/rewire/blob/master/lib/getDefinePropertySrc.js

**Comment by speedskater** [comment]
@TheSavior thanks for reporting this issue and generating the unit test. I will have a look at it on thursday and add a fix to babel-plugin-rewire.

**Comment by TheSavior** [comment]
Thank you very much, this is the last thing keeping us from being able to use Babel internally. I spent some time looking at fixing it myself but got lost in the Babel transforms. :)

It seems like there are a few different ways to export modules with es6, do we need to be adding tests for all of those cases? Being new to es6, I'm not even sure what they all are haha.

**Comment by TheSavior** [comment]
Any update on this? We are really excited to be able to switch over. :)

**Comment by speedskater** [comment]
@TheSavior Sorry for the delay, but i have been really busy over the last days. I try to create the fix asap.

**Comment by TheSavior** [comment]
Thank you. :) 

Also, is there any documentation that you know of on this babel type / transform API stuff? I know it is coming from the Acorn parser and ESTree spec but I'm trying to figure out what the arguments to `t.callExpression` can be, for example.

**Comment by speedskater** [comment]
Hi. 
Sorry for the delay but your issue should be resolved in 0.1.13. 
Regarding the transform API, i think the babel types API correspond to  the spider monkey parser api: https://developer.mozilla.org/en-US/docs/Mozilla/Projects/SpiderMonkey/Parser_API
@TheSavior Could you please let me know if the fix works for you?

**Comment by TheSavior** [comment]
Thank you very much for working on this. It looks like the code doesn't cover all the different ways code is exported
